hack/gen-crd.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -euo pipefail
+
+function install_crd {
+  local SRC="$1"
+  local DST="$2"
+  if ! diff -Naup "$SRC" "$DST"; then
+    cp "$SRC" "$DST"
+    echo "installed CRD: $SRC => $DST"
+  fi
+}
+
+# Generate manifests
+go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths=./... output:crd:artifacts:config=config/crd/bases
+sed -i '/^    controller-gen.kubebuilder.io\/version: (devel)/d' config/crd/bases/*
+
+# Copy crds into the manifests
+install_crd \
+  "config/crd/bases/metal3.io_provisionings.yaml" \
+  "manifests/0000_30_cluster-baremetal-operator_00_metal3provisioning.crd.yaml"
```

Comment:
Continuing from comment above. here as an example, I would put it in install/0000_30... Because ultimately all the installation manifests (service accounts, rbac, etc) end up here...

Reply:
Do we want 2 copies of metal3provisioning.crd.yaml in the repo? At line 15, we could directly add the generated CRD to the (install/manifests)/0000_30_cluster-baremetal-operator_00_metal3provisioning.crd.yaml.

Reply:
I think it depends on how much additional manipulation is required after the raw generation. In such case I noticed that there was an initial replace, that I kept as it was defined, and the renaming part. In general probably we could split into two separate actions (generate raw CRD _and_ copy them into the manifest folder)

Reply:
I am in favor of two copies as well (generated and copy/renamed one).

Reply:
Lets keep the 2 copies. At this point, we don't have any modifications that we need to make to the generated CRD. So, OK to just copy it over.